honeycomb-core/src/orbits.rs

```diff
@@ -0,0 +1,267 @@
+//! Orbit implementation
+//!
+//! This module contains all code used to model orbits, a notion defined
+//! along the structure of combinatorial maps.
+
+// ------ IMPORTS
+
+use crate::{CoordsFloat, DartIdentifier, TwoMap, NULL_DART_ID};
+use num::Zero;
+use std::collections::{BTreeSet, VecDeque};
+
+// ------ CONTENT
+
+/// Enum used to model beta functions defining the search.
+///
+/// This is used to define special cases of orbits that are often used in
+/// algorithms. These special cases correspond to *i-cells*.
+pub enum OrbitPolicy<'a> {
+    /// 0-cell orbit.
+    Vertex,
+    /// 1-cell orbit.
+    Edge,
+    /// 2-cell orbit.
+    Face,
+    /// User-defined orbit. The integers correspond to the indices of the
+    /// beta functions that should be used for the computation.
+    Custom(&'a [u8]),
```

Comment:
```suggestion
    /// Ordered array of beta functions that define the orbit
    Custom(&'a [u8]),
```

Not sure it is mathematically correct, but I think the order thing is important.

Reply:
This is actually a much better description!